runtime/cudaq.h

```diff
@@ -96,25 +91,17 @@ std::string get_kernel_template_member_name() {
 }
 
 /// Get the name of a plain old function that is marked as a quantum kernel.
-inline std::string get_kernel_function_name(std::string &&name) {
+inline std::string get_kernel_function_name(std::string name) {
   return "function_" + std::move(name);
 }
-inline std::string get_kernel_function_name(const std::string &name) {
-  return "function_" + name;
-}
```

Comment:
These changes should be reverted.

The first overload is the r-value reference overload where std::move() makes perfect sense. The second overload covers cases where there isn't an r-value reference and, more importantly, doesn't make a copy.

This change forces a tax on every call to make a copy of the string first.